examples/imagenet/train_imagenet.py

```diff
@@ -24,6 +25,78 @@
 import resnet50
 import resnext50
 
+try:
+    from nvidia import dali
+    from nvidia.dali import pipeline
+    from nvidia.dali import ops
+    _dali_available = True
+except ImportError:
+    _dali_available = False
+
+from chainer.backends import cuda
+import ctypes
+
+
+def _pair(x):
+    if hasattr(x, '__getitem__'):
+        return x
+    return x, x
+
+
+class ImagenetDaliPipeline(pipeline.Pipeline):
+
+    def __init__(self, file_list, file_root, crop_size,
+                 batch_size, num_threads, device_id,
+                 random_shuffle=False, seed=-1,
+                 mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+                 std=[0.229 * 255, 0.224 * 255, 0.225 * 255]):
+        super(ImagenetDaliPipeline, self).__init__(batch_size, num_threads,
+                                                   device_id, seed=seed)
+        crop_size = _pair(crop_size)
+        self.loader = ops.FileReader(file_root=file_root, file_list=file_list,
+                                     random_shuffle=random_shuffle)
+        self.decode = ops.HostDecoder()
+        self.rrcrop = ops.RandomResizedCrop(device="gpu", size=crop_size)
+        self.cmnorm = ops.CropMirrorNormalize(
+            device="gpu", crop=crop_size, mean=mean, std=std)
+        self.coin = ops.CoinFlip(probability=0.5)
+
+    def define_graph(self):
+        jpegs, labels = self.loader()
+        images = self.decode(jpegs)
+        images = self.rrcrop(images.gpu())
+        images = self.cmnorm(images, mirror=self.coin())
+        return [images, labels]
+
+
+def dali_converter(inputs, device=None):
```

Comment:
I think it's better to provide this feature in the framework, rather than in an example.
I even think it's possible to merge this into existing `concat_examples`, but wondering if it is too aggressive... (how do you think? @okuta)

Reply:
I agree with you that it is better to put `dali_converter` in the framework. But, I'm not sure whether it is good idea to merge this feature into `concat_examples`.